app/src/main/java/org/jellyfin/androidtv/data/service/BackgroundService.kt

```diff
@@ -174,6 +175,48 @@ class BackgroundService(
 		}
 	}
 
+	/**
+	 * Use one available backdrop from [SearchHint] as background.
+	 */
+	fun setBackground(searchHint: SearchHint?) {
+		// Check if item is set and backgrounds are enabled
+		if (searchHint == null || !userPreferences[UserPreferences.backdropEnabled])
+			return clearBackgrounds()
+
+		//Manually grab the backdrop URL
+		val options = ImageOptions()
+		options.maxWidth = 1200
+		options.imageType = ImageType.Backdrop
+		options.tag = searchHint.backdropImageTag
+
+		val backdropUrl = apiClient.GetImageUrl(searchHint.backdropImageItemId, options)
+
+		if (backdropUrl.isNullOrEmpty()) return clearBackgrounds()
```

Comment:
It seems like all of the code from this point on is common with the other `setBackground` ~method~ function. It would probably be best to create a private function for this that just accepts the url string.